code/obj/item/cigarette.dm

```diff
@@ -1155,7 +1160,7 @@
 			var/turf/T = get_turf(src.loc)
 			if (T)
 				T.hotspot_expose(700,5)
-			if (fuel == 0)
+			if (reagents.get_reagent_amount("fuel") == 0)
```

Comment:
```suggestion
			if (!reagents.get_reagent_amount("fuel"))
```

Reply:
That should do it. Added a reagents check as well. Thanks for all the help! Hopefully this will be the last time you have to review this ^^;